crystal_toolkit/components/cohp.py

```diff
@@ -0,0 +1,781 @@
+            n_bins=256,
+            bva_comp=True,
+        )
+        calc_quality_description = Description.get_calc_quality_description(
+            calc_quality_dict
+        )
+
+        return " ".join(calc_quality_description)
+
+    @staticmethod
+    def get_lobster_local_envs(
+        charge_obj, completecohp_obj, icohplist_obj, madelung_obj, which_bonds="all"
+    ) -> str:
+        """Get text description of local environments
+
+        Args:
+            input_dict: Dictionary containing the pymatgen objects.
+
+        Returns:
+            A string describing the local environments.
+        """
+        # Get the local environments using LobsterPy
+        analyse = Analysis(
+            charge_obj=charge_obj,
+            madelung_obj=madelung_obj,
+            icohplist_obj=icohplist_obj,
+            completecohp_obj=completecohp_obj,
+            path_to_poscar=None,
+            path_to_icohplist=None,
+            path_to_cohpcar=None,
+            which_bonds=which_bonds,
+            summed_spins=False,
+        )
+
+        envs = []  # list of local environments
+        for site_ix, env in enumerate(analyse.lse.coordination_environments):
+            if site_ix in analyse.seq_ineq_ions and env[0]["ce_symbol"]:
+                # if env[0]["ce_symbol"]:
+                data_list = []
+                site_str = unicodeify_species(analyse.structure[site_ix].species_string)
+
+                try:
+                    data_list.extend(
+                        [
+                            ["Site", site_str],
+                            [
+                                "Environment",
+                                Description._coordination_environment_to_text(
+                                    env[0]["ce_symbol"]
+                                ).capitalize(),
+                            ],
+                            ["IUPAC Symbol", env[0]["ce_symbol"]],
+                            ["CSM", float(round(env[0]["csm"], 5))],
+                        ]
+                    )
+
+                except KeyError:
+                    data_list.extend(
+                        [
+                            ["Site", site_str],
+                            [
+                                "Environment",
+                                Description._coordination_environment_to_text(
+                                    env[0]["ce_symbol"]
+                                ).capitalize(),
+                            ],
+                            ["IUPAC Symbol", env[0]["ce_symbol"]],
+                            ["CSM", "NA"],
+                        ]
+                    )
+
+                local_env_data = analyse.chemenv.get_nn_info(analyse.structure, site_ix)
+
+                neighbour_sites = [i["site"] for i in local_env_data]
+                central_site = analyse.structure[site_ix]
+                neighbour_weights = [
+                    i["edge_properties"]["ICOHP"] for i in local_env_data
```

Comment:
I am not sure you can simply assign this to each bond by order. If you remember, there might have been a bug here. This is something to recheck.